src/components/SuperDrawer/useSuperDrawer.tsx

```diff
@@ -64,40 +64,48 @@ export function useSuperDrawer(): UseSuperDrawerHook {
   } = useContext(BeamContext);
   const { openModal } = useModal();
 
+  /**
+   * Validates whether a new Superdrawer can be opened / whether the current Superdrawer can be closed.
+   * Breaking the validation into a separate function allows openInDrawer
+   * to validate the drawer's state without closing it
+   * */
+  function validateDrawerChange (onClose: () => void) {
+    const contentStackLength = contentStack.current.length;
+    if (contentStackLength === 0) {
+      return true;
+    }
+
+    // Attempt to close each drawer details
+    for (let i = contentStackLength - 2; i >= 0; i--) {
+      for (const canCloseDrawerDetail of canCloseDetailsChecks.current[i] ?? []) {
+        if (!canCloseDrawerDetail()) {
+          openModal({
+            title: "Confirm",
+            content: <ConfirmCloseModal onClose={onClose} />,
+          });
+          return false;
+        }
+      }
+    }
+
+    // Attempt to close the drawer
+    for (const canCloseDrawer of canCloseChecks.current) {
+      if (!canCloseDrawer()) {
+        openModal({
+          title: "Confirm",
+          content: <ConfirmCloseModal onClose={onClose} />,
+        });
+        return false;
+      }
+    }
+  }
+
+
   // Separate close actions to reference then in actions
   const closeActions = useMemo(() => {
     return {
       /** Attempts to close the drawer. If any checks fail, a error modal will appear */
```

Comment:
While you're here, maybe change `error modal` to `confirmation modal` on the line above?